hack/jenkins/test-history/gen_html.py

```diff
@@ -84,129 +90,191 @@ def gen_tests(data, prefix, exact_match):
             else:
                 status = 'okay'
             test_html.append('<li class="suite">')
-            test_html.append('<span class="%s">%d/%d</span>' % (status, num_builds - num_failed, num_builds))
-            test_html.append('<span class="time">%.0f%s</span>' % (avg_time, unit))
+            test_html.append('<span class="%s">%d/%d</span>' % (
+                status, num_builds - num_failed, num_builds))
+            test_html.append(
+                '<span class="time">%.0f%s</span>' % (avg_time, unit))
             test_html.append(suite)
             test_html.append('</li>')
         test_html.append('</ul>')
         if has_failed:
             status = 'failed'
-            total_failed += 1
         elif has_unstable:
             status = 'unstable'
-            total_unstable += 1
         elif has_test:
             status = 'okay'
-            total_okay += 1
         else:
             status = 'skipped'
-            total_skipped += 1
+        totals[status] += 1
         html.append('<li class="test %s">' % status)
         if exact_match and len(test_html) > 2:
-            if not (test_html[2].startswith('<span') and test_html[3].startswith('<span')):
-                raise ValueError("couldn't extract suite results for prepending")
+            if not (test_html[2].startswith('<span') and
+                    test_html[3].startswith('<span')):
+                raise ValueError(
+                    'couldn\'t extract suite results for prepending')
             html.extend(test_html[2:4])
             html.append(test)
         else:
             html.append(test)
             html.extend(test_html)
         html.append('</li>')
     html.append('</ul>')
-    return '\n'.join(html), (total_okay, total_unstable, total_failed, total_skipped)
+    return '\n'.join(html), TestMetadata(
+        totals['okay'], totals['unstable'], totals['failed'], totals['skipped'])
+
 
 def html_header():
+    """Return html header items."""
     html = ['<html>', '<head>']
     html.append('<link rel="stylesheet" type="text/css" href="style.css" />')
     html.append('<script src="script.js"></script>')
```

Comment:
I know this isn't introduced by you, but this script shouldn't be in `index.html`. Could you move this line into `gen_html`, probably right before the `</body>` line?

It makes an ugly javascript error that causes the very first bubble to be greyed out.